### PR TITLE
Drop Scala 2.12 support

### LINKED ISSUE #11315: Drop Scala 2.12 support

### PR DESCRIPTION
**Pull request by cocreature**
This only includes the CI/build system infrastructure
changes. Dropping compatibility layers from our code for 2.12 can be
done separately.

This is fine even in the context of backport builds since we already
disable the Scala 2.12 job for those anyway.

fixes #11315

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: BAZEL.md [hunk 1]
```diff
@@ -1033,10 +1033,9 @@ All direct Scala and Java dependencies are listed explicitly in the file
 `maven_install` repository rule calls Coursier to perform transitive dependency
 resolution and import the required artifacts into the Bazel build.
 
-The resolved versions are pinned in the file `maven_install.json`. Execute
+The resolved versions are pinned in the file `maven_install_2.13.json`. Execute
```
Comment (SamirTalwar-DA) [inline BAZEL.md:1036]: Are we keeping this because we expect to upgrade to Scala 3.x at some point?

**Review by SamirTalwar-DA — APPROVED** [review]
Nice work!

**Comment by SamirTalwar-DA** [comment]
Are you planning on purging `versioned_scala_deps` in a separate PR?

File: BAZEL.md (hunk 1 above)
Comment (cocreature) [reply BAZEL.md:1036]: I hope so yeah. And it doesn’t really seem to cause any harm to have the suffix.

**Comment by cocreature** [comment]
> Are you planning on purging `versioned_scala_deps` in a separate PR?

yeah I tried to keep this to the stuff that shouldn’t conflict with anything so no changes to individual targets unless strictly needed.

File: bazel-java-deps.bzl [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # When adding, removing or changing a dependency in this file, update the pinned dependencies by executing
-# $ bazel run @unpinned_maven//:pin && DAML_SCALA_VERSION=2.12.14 bazel run @unpinned_maven//:pin
+# $ bazel run @unpinned_maven//
```
Comment (pbatko-da) [inline bazel-java-deps.bzl:5]:
```suggestion
# $ bazel run @unpinned_maven//pin
```
?

Comment (cocreature) [reply bazel-java-deps.bzl]: good catch

**Pull request merged**